api/types/model.go

```diff
@@ -179,3 +171,27 @@ type SourceContext struct {
 	MountPath string        `json:"mountPath"`
 	Context   io.ReadCloser `json:"-"`
 }
+
+func FilterKeysWithPublicKey(keys []SSHKey) []SSHKey {
+	filteredKeys := make([]SSHKey, 0)
+
+	for _, key := range keys {
+		if key.PublicKey != nil {
+			filteredKeys = append(filteredKeys, key)
+		}
+	}
+
+	return filteredKeys
+}
+
+func GetPublicKeys(keys []SSHKey) []string {
+	filteredStrings := make([]string, 0)
+
+	for _, key := range keys {
+		if key.PublicKey != nil {
+			filteredStrings = append(filteredStrings, *key.PublicKey)
+		}
+	}
+
+	return filteredStrings
+}
```

Comment:
Tbh I'm not a huge fan of these utility functions. If I have to read the function to understand what it's doing everything, it defeats the purpose of it. Unless we're using this functionality my times over, I don't think we should be putting it in a top level types package. We can define local private methods where this functionality is needed (unless it's being used repeatedly).

Reply:
I think the point is so you can read the function to understand what it’s doing by the combination of function names alone. I personally think this is much better than having to grock actual code in Golang every single time, because then you have to approximate the behaviour of the call site which takes more time than just reading function names.